cups/ppd-private.h

```diff
@@ -101,6 +101,16 @@ typedef enum _pwg_print_quality_e	/**** PWG print-quality values ****/
   _PWG_PRINT_QUALITY_MAX
 } _pwg_print_quality_t;
 
+typedef enum _pwg_print_content_optimize_e /** PWG print-content-optimize **/
+{
+  _PWG_PRINT_CONTENT_OPTIMIZE_AUTO = 0, /* print-content-optimize=auto */
+  _PWG_PRINT_CONTENT_OPTIMIZE_PHOTO,	/* print-content-optimize=photo */
+  _PWG_PRINT_CONTENT_OPTIMIZE_GRAPHICS, /* print-content-optimize=graphics */
```

Comment:
"graphic" (singular). Some printers incorrectly report "graphics".

That said, I think we should only set this in presets for "photo". Anything else isn't even vaguely useful.

Reply:
OK, I have made the part of CUPS which executes jobs and there applies the presets understand both, as I was in doubt. So using singular everywhere and CUPS only understanding singular is good enough as this is the standard? The plural can simply have come from my understanding of English, assuming that one says "This document is graphics" and not "This document is graphic".

Reply:
@tillkamppeter The PWG standard uses singular keywords while the normal English usage allows for either. And unfortunately there are printers out there that use the plural form due to a typo in an older AirPrint specification that was later fixed. So you *do* need to support both... Ideally I would just map "print-content-optimize" to "cupsPrintOptimize" and then support "photo" for the photo presets and ignore the rest for APPrinterPreset.

Reply:
So we leave it the way that we accept both singular and plural of "graphics"?

But "cupsPrintOptimize" is only in the generated PPDs for driverless IPP printers? My intention is also to support existing PPDs for non-driverless printers as well as possible.

What do you mean with 'and then support "photo" for the photo presets and ignore the rest for APPrinterPreset.'? Where are "photo" presets? I only see the grid of 6 presets for the combos of mono/color and draft/normal/high quality.